### DISCUSSION TITLE
Does the RuleEngine evaluate rules concurrently or sequentially?

### DISCUSSION THREAD

**Discussion by anshulk-arch** [microsoft/RulesEngine#426] [Q&A]
In a given workflow with N rules, does the RuleEngine execute these rules concurrently or sequentially from 1 to N e.g. in the workflow below, will it execute GiveDiscount10 and GiveDiscount20 in parallel or, one after the other? If sequential, is it possible to execute rules sequentially?

[
  {
    "WorkflowName": "Discount",
    "Rules": [
      {
        "RuleName": "GiveDiscount10",
        "SuccessEvent": "10",
        "ErrorMessage": "One or more adjust rules failed.",
        "ErrorType": "Error",
        "RuleExpressionType": "LambdaExpression",
        "Expression": "input1.country == \"india\" AND input1.loyaltyFactor <= 2 AND input1.totalPurchasesToDate >= 5000"
      },
      {
        "RuleName": "GiveDiscount20",
        "SuccessEvent": "20",
        "ErrorMessage": "One or more adjust rules failed.",
        "ErrorType": "Error",
        "RuleExpressionType": "LambdaExpression",
        "Expression": "input1.country == \"india\" AND input1.loyaltyFactor >= 3 AND input1.totalPurchasesToDate >= 10000"
      }
    ]
  }
]

**Comment by mtelang** [comment]
As per my understanding, it does sequentially. So if there are two rules in one workflow which fulfill the condition, the execution will stop after the first one

**Comment by Sanjay-kumar465** [comment]
[Uploading audit_history_1780031488711.xlsx…]()
